src/tools/miri/tests/pass/shims/fs.rs

```diff
@@ -32,14 +32,18 @@ fn main() {
 }
 
 fn host_to_target_path(path: String) -> PathBuf {
-    use std::ffi::{CStr, CString};
+    use std::ffi::{c_char, CStr, CString};
 
     let path = CString::new(path).unwrap();
     let mut out = Vec::with_capacity(1024);
 
     unsafe {
         extern "Rust" {
-            fn miri_host_to_target_path(path: *const i8, out: *mut i8, out_size: usize) -> usize;
+            fn miri_host_to_target_path(
+                path: *const c_char,
+                out: *mut c_char,
+                out_size: usize,
+            ) -> usize;
```

Comment:
AArch64 uses unsigned char while x86 uses signed char. Using c_char works on both.

Reply:
Ah, I've been looking at https://doc.rust-lang.org/nightly/alloc/ffi/struct.CString.html#method.as_ptr which claims the return type is an `i8` pointer but looking at the source shows that rustdoc is omitting some crucial information here.